projects/storefrontlib/layout/main/storefront.component.ts

```diff
@@ -121,6 +134,16 @@ export class StorefrontComponent implements OnInit, OnDestroy {
     this.hamburgerMenuService.toggle(true);
   }
 
+  excludeFromFocus(isExpanded: boolean): void {
+    const tabindex = isExpanded ? '-1' : '0';
+    const logo = this.elementRef.nativeElement.querySelector('cx-banner a');
```

Comment:
I think your approach of applying `tabindex` to elements is acceptable as I can not think of how we can apply the behaviour correctly otherwise given that the layout is delivered to us from the backend a specific way.

We should keep in mind that user's may have (and likely do) different page slot and layout configurations than us. We need to keep this under consideration with our solution.

In an ideal world, we would have the navigation and the close button in the same slot so that we don't need to modify elements outside of it. We could also use the `dialog` element which mostly has all of the a11y keyboard requirements (ie. tab trap) we need built in.

With all that said, we should consider what we do have and what might solution might cover all edge cases. I see that in particular, section `navigation` needs to all be focusable as well as the `close` button outside. And that's it.
Is it possible to select all elements that could be focusable outside of this criteria (`button` or `a` elements that are visible)? If we can block those indiscriminately with `tabindex`, we don't need to worry as much if they have their page structured as much as we expect.